sql/src/main/java/io/crate/operation/projectors/ShardRequestAccumulator.java

```diff
@@ -110,6 +126,24 @@ public int batchSize() {
         return listener;
     }
 
+    @Nullable
+    private String getDestinationNodeId() {
+        ShardIterator shardIterator = clusterService.operationRouting().indexShards(clusterService.state(),
```

Comment:
I think instead of re-retrieving the target node we should just assume that it is the node this operation is running on. Because a `shardId` is already routed. Of course there is a chance that the shard has been relocated and that the requests will re-route themselves to the new node. But there is always the race condition for that to happen. So I think the accounting should only consider the "initial target". We also always have replica requests and don't account for them.